### PR TITLE
use 'experiment' prop instead of 'getExperimentBySlug' on ExperimentPage

### PR DESCRIPTION
**Pull request by dannycoates**
This feels cleaner to me...

### CODE REVIEW

File: frontend/src/app/containers/App.js [hunk 1]
```diff
@@ -59,17 +59,16 @@ function subscribeToBasket(email, callback) {
...
     experiments: experimentSelector(state),
     isFirefox: state.browser.isFirefox,
     isMinFirefox: state.browser.isMinFirefox,
     isDev: state.browser.isDev,
     hasAddon: state.addon.hasAddon,
     installed: getInstalled(state.addon),
     installedAddons: state.addon.installedAddons,
-    getExperimentBySlug: slug =>
-      getExperimentBySlug(state.experiments, slug),
     isExperimentEnabled: experiment =>
```
Comment (dannycoates) [inline frontend/src/app/containers/App.js:73]: you're next 🎯

File: frontend/src/app/containers/App.js [hunk 2]
```diff
@@ -59,17 +59,16 @@ function subscribeToBasket(email, callback) {
 }
 
 export default connect(
-  state => ({
+  (state, ownProps) => ({
     addon: state.addon,
+    experiment: getExperimentBySlug(state.experiments, ownProps.params.slug),
```
Comment (lmorchard) [inline frontend/src/app/containers/App.js]: I'm kind of leary of this solution. Right now, `/experiment/` is the only page that uses `:slug` in the router. But, that might not always be true and this lookup is specific to the experiment details page. This has us looking up the experiment for _every_ page, whether or not it needs one or has a slug in the URL.

I've been wondering if maybe a cleaner solution would be to put a wrapper around the `ExperimentPage` component in the router that does the experiment lookup & passes it in as a prop. That way, the experiment lookup logic would be limited to the router & the single route.

Comment (lmorchard) [reply frontend/src/app/containers/App.js]: By wrapper, I mean something like this (which I haven't tried):

``` javascript
      <Route path="/experiments/:slug" 
             component={props => <ExperimentPage {...props} experiment={props.getExperimentBySlug(props.params.slug)}} /> } />
```

Comment (dannycoates) [reply frontend/src/app/containers/App.js]: Thanks @lmorchard, I didn't know you could do that 😎

**Review by lmorchard — APPROVED** [review]

**Pull request merged**